src/main/java/ru/testtask/service/FileService.java

```diff
@@ -0,0 +1,147 @@
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public List<Path> getFileListFromDirectory(Path path) throws IOException {
+
+        List<Path> paths = new ArrayList<>();
+
+        Files.walkFileTree(path, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                if (!Files.isDirectory(file))
+                paths.add(file);
+                return super.visitFile(path, attrs);
+            }
+        });
+
+        return paths;
+    }
+
+    public void removeFile(String filename){
+        fileDataRepo.deleteByFilename(filename);
+    }
+
+    public FileData findByFilename(String filename) throws IOException {
+        Files.walkFileTree(Paths.get(uploadPath), new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                if (file.getFileName().toString().equals(filename))
+                    Files.delete(file);
+                return super.visitFile(file, attrs);
+            }
+        });
+        return fileDataRepo.findByFilename(filename);
+    }
+
+    public List<FileData> searchByRegex(String regex){
+        List<FileData> files = fileDataRepo.findAll();
+        List<FileData> appropriateFiles = new ArrayList<>();
+
+        for (FileData fileData : files){
+            if (fileData.getFilename().matches("(.*)" + regex + "(.*)"))
+                appropriateFiles.add(fileData);
+        }
+
+        return appropriateFiles;
+    }
+
+    public void moveFile(String filename, Path destination) throws IOException {
+
+        if (!Files.exists(destination)){
+            Files.createDirectory(destination);
+        }
+
+        Files.walkFileTree(Paths.get(uploadPath), new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                if (file.getFileName().toString().equals(filename)) {
+                    Files.move(file, destination.resolve(file.getFileName()));
+                }
+                return super.visitFile(file, attrs);
+            }
+        });
+
+        FileData fileData = fileDataRepo.findByFilename(filename);
+        fileData.setDirectory(destination.toString());
+        fileDataRepo.save(fileData);
+    }
+
+    public void copyFile(String filename, Path destination) throws IOException {
+        String newFilename;
+        int i = 0;
+
+        if (!Files.exists(destination)){
+            Files.createDirectory(destination);
+        }
+
+        FileData fileData = fileDataRepo.findByFilename(filename);
+
+        do {
+            newFilename = fileData.getFilename() + "(" + i + ")";
+            i++;
+        }
+        while (fileDataRepo.findByFilename(newFilename) != null);
```

Comment:
It isn't good idea to make requests in each iteration. And this way does not guarantee to succesfully creation the file